dev/TabView/TabViewItem.cpp

```diff
@@ -97,7 +96,33 @@ void TabViewItem::OnApplyTemplate()
 
     UpdateCloseButton();
     UpdateForeground();
-    UpdateWidthModeVisualState();    
+    UpdateWidthModeVisualState();
+    UpdateTabGeometry();
+}
+
+void TabViewItem::UpdateTabGeometry()
+{
+    auto const templateSettings = winrt::get_self<TabViewItemTemplateSettings>(TabViewTemplateSettings());
+
+    auto const height = ActualHeight();
+    auto const popupRadius = unbox_value<winrt::CornerRadius>(ResourceAccessor::ResourceLookup(*this, box_value(c_overlayCornerRadiusKey)));
+    auto const leftCorner = popupRadius.TopLeft;
+    auto const rightCorner = popupRadius.TopRight;
+
+    // Assumes 4px curving-out corners, which are hardcoded in the markup
```

Comment:
Are these really hardcoded? Couldn't the 4px be customized in XAML markup?

Reply:
It's the width/height of the LeftRadiusRenderArc and RightRadiusRenderArc, which I guess I could measure... the string is getting very unreadable/undebuggable, though. Is there a better way to construct this?

Reply:
Let's leave it as is for now, and if a request for customizability comes up, we can investigate at that point.